pkg/generate/code/late_intialize_test.go

```diff
@@ -0,0 +1,136 @@
+
+	crd := testutil.GetCRDByName(t, g, "Repository")
+	require.NotNil(crd)
+	// FieldConfig without lateInitialize
+	assert.NotEmpty(crd.Config().ResourceFields(crd.Names.Original)["Name"])
+	assert.Nil(crd.Config().ResourceFields(crd.Names.Original)["Name"].LateInitialize)
+	assert.Empty(code.FindLateInitializedFieldsWithDelay(crd.Config(), crd, 1))
+}
+
+func Test_FindLateInitializedFieldsWithDelay(t *testing.T) {
+	assert := assert.New(t)
+	require := require.New(t)
+
+	g := testutil.NewModelForServiceWithOptions(t, "ecr", &testutil.TestingModelOptions{GeneratorConfigFile: "generator-with-late-initialize.yaml"})
+
+	crd := testutil.GetCRDByName(t, g, "Repository")
+	require.NotNil(crd)
+	// FieldConfig without lateInitialize
+	assert.NotEmpty(crd.Config().ResourceFields(crd.Names.Original)["Name"])
+	assert.NotEmpty(crd.Config().ResourceFields(crd.Names.Original)["ImageTagMutability"])
+	assert.NotNil(crd.Config().ResourceFields(crd.Names.Original)["Name"].LateInitialize)
+	assert.NotNil(crd.Config().ResourceFields(crd.Names.Original)["ImageTagMutability"].LateInitialize)
+	expected := "\tvar lateInitializeFieldToDelaySeconds = map[string]int{\"ImageTagMutability\":5,\"Name\":0,}\n"
+	assert.Equal(expected, code.FindLateInitializedFieldsWithDelay(crd.Config(), crd, 1))
+}
+
+func Test_LateInitializeFromReadOne_NoFieldsToLateInitialize(t *testing.T) {
+	assert := assert.New(t)
+	require := require.New(t)
+
+	g := testutil.NewModelForService(t, "ecr")
+
+	crd := testutil.GetCRDByName(t, g, "Repository")
+	require.NotNil(crd)
+	// NO fieldConfig
+	assert.Empty(crd.Config().ResourceFields(crd.Names.Original))
+	assert.Empty(code.LateInitializeFromReadOne(crd.Config(), crd, "observed", "koWithDefaults", 1))
+}
+
+func Test_LateInitializeFromReadOne_NonNestedPath(t *testing.T) {
+	assert := assert.New(t)
+	require := require.New(t)
+
+	g := testutil.NewModelForServiceWithOptions(t, "ecr", &testutil.TestingModelOptions{GeneratorConfigFile: "generator-with-late-initialize.yaml"})
+
+	crd := testutil.GetCRDByName(t, g, "Repository")
+	require.NotNil(crd)
+	// FieldConfig without lateInitialize
+	assert.NotEmpty(crd.Config().ResourceFields(crd.Names.Original)["Name"])
+	assert.NotEmpty(crd.Config().ResourceFields(crd.Names.Original)["ImageTagMutability"])
+	assert.NotNil(crd.Config().ResourceFields(crd.Names.Original)["Name"].LateInitialize)
+	assert.NotNil(crd.Config().ResourceFields(crd.Names.Original)["ImageTagMutability"].LateInitialize)
+	expected :=
+`	if observed.Spec.ImageTagMutability != nil && koWithDefaults.Spec.ImageTagMutability == nil {
+		koWithDefaults.Spec.ImageTagMutability = observed.Spec.ImageTagMutability
+	}
+	if observed.Spec.Name != nil && koWithDefaults.Spec.Name == nil {
+		koWithDefaults.Spec.Name = observed.Spec.Name
+	}
+`
+	assert.Equal(expected, code.LateInitializeFromReadOne(crd.Config(), crd, "observed", "koWithDefaults", 1))
+}
+
+func Test_LateInitializeFromReadOne_NestedPath(t *testing.T) {
```

Comment:
Thinking that maybe we can add more unit tests for a "nested struct in maps in structs"